com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/configuration/UAAClientConfiguration.java

```diff
@@ -34,7 +34,7 @@ private URL readTokenEndpoint(URL targetURL) {
         try {
             String infoURL = targetURL.toString() + "/v2/info";
             ResponseEntity<String> infoResponse = new RestTemplate().getForEntity(infoURL, String.class);
-            if (infoResponse == null) {
+            if (infoResponse.getBody() == null) {
```

Comment:
Can the response itself be `null`?

Reply:
As far as I observed the code of the Spring, the response is always built. However, based on the Response type(in this case String.class), the body is decided whether to be included in the response or not.
You could check ResponseEntityResponseExtractor which is component responsible for getting the response body out of the HTTP response